client/reader/stream/index.jsx

```diff
@@ -365,9 +449,13 @@ export default class ReaderStream extends React.Component {
 	}
 
 	renderPost = ( postKey, index ) => {
-		let postState,
-			PostClass;
-		const isSelected = index === this.state.selectedIndex;
+		const selectedPostKey = this.props.store.getSelectedPost();
+		const isSelected = selectedPostKey &&
+			selectedPostKey.postId === postKey.postId &&
+			(
```

Comment:
we have two different sources of posts. posts from a feed have a feed id, posts from a site have a blog id. so anything from /read/following or /read/feeds/:feed/* will have a feed id, while anything from /read/blogs/* or /read/tag/* or search will generally have a blog id.